### PR TITLE
Refactor: Migrate LOI models to Kotlin

### PR DESCRIPTION
**Pull request by scolsen**
Migrates the LOI classes to Kotlin. We define temporary builders to
prevent breaking existing Java callers.

Furthermore, this commit renames PolygonOfInterest->AreaOfInterest and
updates names for fake test data.

### CODE REVIEW

**Comment by scolsen** [comment]
lol, I have yet to open a PR that didn't fail on the first run thanks to import order problems...should be fixed now

File: ground/src/main/java/com/google/android/ground/model/TestModelBuilders.java [hunk 1]
```diff
@@ -17,9 +17,9 @@
 package com.google.android.ground.model;
 
 import com.google.android.ground.model.job.Job;
+import com.google.android.ground.model.locationofinterest.AreaOfInterest;
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/model/TestModelBuilders.java:20]: Unrelated: I'd be in favor of using `loi` for package and variable names.

**Review by gino-m — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/model/TestModelBuilders.java (hunk 1 above)
Comment (scolsen) [reply ground/src/main/java/com/google/android/ground/model/TestModelBuilders.java:20]: sgtm

Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/model/TestModelBuilders.java:20]: Also, let's discuss how we can consolidate our representation of point, polygon, and geojson in code and in the db.

**Pull request merged**